packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -475,30 +478,40 @@ export class AutoScalingGroup extends AutoScalingGroupBase implements
 
     launchConfig.node.addDependency(this.role);
 
-    const desiredCapacity =
-        (props.desiredCapacity !== undefined ? props.desiredCapacity :
-        (props.minCapacity !== undefined ? props.minCapacity :
-        (props.maxCapacity !== undefined ? props.maxCapacity : 1)));
+    // desiredCapacity just reflects what the user has supplied.
+    const desiredCapacity = props.desiredCapacity;
     const minCapacity = props.minCapacity !== undefined ? props.minCapacity : 1;
-    const maxCapacity = props.maxCapacity !== undefined ? props.maxCapacity : desiredCapacity;
+    const maxCapacity = props.maxCapacity !== undefined ? props.maxCapacity :
+                        desiredCapacity !== undefined ? desiredCapacity : Math.max(minCapacity, 1);
 
+    withResolved(minCapacity, maxCapacity, (min, max) => {
+      if (min > max) {
+        throw new Error(`minCapacity (${min}) should be <= maxCapacity (${max})`);
+      }
+    });
     withResolved(desiredCapacity, minCapacity, (desired, min) => {
+      if (desired === undefined) { return; }
       if (desired < min) {
         throw new Error(`Should have minCapacity (${min}) <= desiredCapacity (${desired})`);
       }
     });
     withResolved(desiredCapacity, maxCapacity, (desired, max) => {
+      if (desired === undefined) { return; }
       if (max < desired) {
         throw new Error(`Should have desiredCapacity (${desired}) <= maxCapacity (${max})`);
       }
     });
 
+    if (desiredCapacity !== undefined) {
+      this.node.addWarning(`desiredCapacity has been configured. Be aware this will reset the size of your AutoScalingGroup on every deployment.`);
```

Comment:
Provider a link to a github issue in the help message so people can dive deeper/comment.